pkg/controllers/node/provisioner_drift.go

```diff
@@ -0,0 +1,33 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"github.com/aws/karpenter-core/pkg/apis/v1alpha5"
+	"github.com/aws/karpenter-core/pkg/scheduling"
+)
+
+func isProvisionerDrifted(machine *v1alpha5.Machine, provisioner *v1alpha5.Provisioner) bool {
```

Comment:
What is the purpose of this function if it simply wraps `isNodeRequirementsDrifted`?

Reply:
The plan is to implements more components of the provisioner drift in the future.  `isProvisionerDrifted ` will be where all the helper function will be located when making individual checks for each provisioner component.

Reply:
Got it, sorry for commenting so quickly on a draft PR!